components/donation-frequency.jsx

```diff
@@ -2,44 +2,45 @@ import React from 'react';
 
 module.exports = React.createClass({
   mixins: [require('react-intl').IntlMixin],
-  getInitialState: function() {
-    return {
-      values: {
-        recurring: 0,
-        description: this.getIntlMessage("mozilla_donation")
-      }
-    };
+  componentDidMount: function() {
+    this.updateFrequency(this.props.value);
   },
   onChange: function(e) {
-    var recurring = e.currentTarget.value;
-    var description = this.state.description;
-    if (recurring) {
+    var value = "single";
+    if (e.currentTarget.value === "1") {
+      value = "monthly";
+    }
+    this.updateFrequency(value);
+  },
+  updateFrequency: function(value) {
```

Comment:
Creating a handy function `updateFrequency` so I can easily use it in the component did mount and the onchange.